RegistryAuthenticator not sending the JIB specific user agent

<!-- Thanks for filing an issue! Please use this template to provide more context for the issue . -->

**Description of the issue**:
We have an internal Artifactory running as docker registry behind Nginx. The setup works well with docker cli. But when I tried to add JIB to our maven project, I ran into a strange "Forbidden" error that was caused by a 404 response code from Nginx. That's not the issue though should be fixed anyway, in another issue maybe. 

After some investigation, I found that our internal Nginx server don't like user agent strings that starts with `Go`. And the `RegistryAuthenticator` happens to send the string `Google-Http-Client ...` in the `authenticate("push,pull")` method. Because it's using google http client and did not set the JIB user agent.

**Expected behavior**:
All HTTP requests sent by JIB should bear the same user agent.

**Steps to reproduce**: <!-- Please provide a minimal and precise series of steps -->
1. You need an Nginx server/registry that rejects HTTP requests based on user agent
2. Execution goal `build`. `dockerBuild` is not affected.
3. Run `mvn package`

**`jib-maven-plugin` Configuration**: <!-- Delete this section if not used -->
```xml
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>jib-maven-plugin</artifactId>
  <version>1.0.2</version>
  <executions>
    <execution>
      <phase>package</phase>
      <goals>
        <goal>build</goal>
      </goals>
    </execution>
  </executions>
  <configuration>
    <from>
      <image>${docker.registry}/test/alpine-python-java:8_server-jre_unlimited</image>
    </from>
    <to>
      <!--suppress UnresolvedMavenProperty -->
      <image>${docker.registry}/test/${project.artifactId}:${docker.image.tag}</image>
      <tags>
        <tag>latest</tag>
      </tags>
    </to>
    <extraDirectory>
      <!--suppress UnresolvedMavenProperty -->
      <path>${main.basedir}/jib</path>
      <permissions>
        <permission>
          <file>/start.sh</file>
          <mode>755</mode>
        </permission>
      </permissions>
    </extraDirectory>
    <container>
      <entrypoint>
        <arg>/start.sh</arg>
        <!--suppress UnresolvedMavenProperty -->
        <arg>${main.class}</arg>
      </entrypoint>
    </container>
  </configuration>
</plugin>
```
**Log output**: <!-- If applicable, provide relevant log output -->
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.0.2:build (default) on project sense-mq: Build image failed, perhaps you should make sure your credentials for 'sense.docker.dev.cba' are set up correctly: Unauthorized for sense.docker.dev.cba/sense/sense-mq: 404 Not Found
[ERROR] <html>
[ERROR] <head><title>404 Not Found</title></head>
[ERROR] <body bgcolor="white">
[ERROR] <center><h1>404 Not Found</h1></center>
[ERROR] <hr><center>nginx/1.13.3</center>
[ERROR] </body>
[ERROR] </html>
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.0.2:build (default) on project sense-mq: Build image failed, perhaps you should make sure your credentials for 'sense.docker.dev.cba' are set up correctly
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Build image failed, perhaps you should make sure your credentials for 'sense.docker.dev.cba' are set up correctly
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:174)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.cloud.tools.jib.registry.RegistryUnauthorizedException: Unauthorized for sense.docker.dev.cba/sense/sense-mq
    at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build (BuildStepsRunner.java:243)
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:132)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.google.api.client.http.HttpResponseException: 404 Not Found
<html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.13.3</center>
</body>
</html>

    at com.google.api.client.http.HttpRequest.execute (HttpRequest.java:1102)
    at com.google.cloud.tools.jib.http.Connection.send (Connection.java:200)
    at com.google.cloud.tools.jib.http.Connection.get (Connection.java:155)
    at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticate (RegistryAuthenticator.java:309)
    at com.google.cloud.tools.jib.registry.RegistryAuthenticator.authenticatePush (RegistryAuthenticator.java:247)
    at com.google.cloud.tools.jib.builder.steps.AuthenticatePushStep.call (AuthenticatePushStep.java:99)
    at com.google.cloud.tools.jib.builder.steps.AuthenticatePushStep.call (AuthenticatePushStep.java:45)
    at com.google.common.util.concurrent.CombinedFuture$CallableInterruptibleTask.runInterruptibly (CombinedFuture.java:180)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
```

**Additional Information**: <!-- Any additional information that may be helpful -->
To manually issue the rejected HTTP request.

```
$ curl -H 'User-Agent: Go' 'https://registry.docker.internal/v2/token?service=registry.docker.internal&scope=repository:test/service:pull,push'
```

<!-- Thanks for contributing! -->

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetriever.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetriever.java
@@ -31,10 +31,12 @@ import javax.annotation.Nullable;
 class AuthenticationMethodRetriever implements RegistryEndpointProvider<RegistryAuthenticator> {
 
   private final RegistryEndpointRequestProperties registryEndpointRequestProperties;
+  private final String userAgent;
 
   AuthenticationMethodRetriever(
-      RegistryEndpointRequestProperties registryEndpointRequestProperties) {
+      RegistryEndpointRequestProperties registryEndpointRequestProperties, String userAgent) {
     this.registryEndpointRequestProperties = registryEndpointRequestProperties;
+    this.userAgent = userAgent;
   }
 
   @Nullable
@@ -96,7 +98,7 @@ class AuthenticationMethodRetriever implements RegistryEndpointProvider<Registry
     // Parses the header to retrieve the components.
     try {
       return RegistryAuthenticator.fromAuthenticationMethod(
-          authenticationMethod, registryEndpointRequestProperties);
+          authenticationMethod, registryEndpointRequestProperties, userAgent);
 
     } catch (RegistryAuthenticationFailedException ex) {
       throw new RegistryErrorExceptionBuilder(getActionDescription(), ex)

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
@@ -131,6 +131,7 @@ public class RegistryAuthenticator {
    *
    * @param authenticationMethod the {@code WWW-Authenticate} header value
    * @param registryEndpointRequestProperties the registry request properties
+   * @param userAgent the {@code User-Agent} header value to use in later authentication calls
    * @return a new {@link RegistryAuthenticator} for authenticating with the registry service
    * @throws RegistryAuthenticationFailedException if authentication fails
    * @see <a
@@ -139,7 +140,8 @@ public class RegistryAuthenticator {
   @Nullable
   static RegistryAuthenticator fromAuthenticationMethod(
       String authenticationMethod,
-      RegistryEndpointRequestProperties registryEndpointRequestProperties)
+      RegistryEndpointRequestProperties registryEndpointRequestProperties,
+      String userAgent)
       throws RegistryAuthenticationFailedException {
     // If the authentication method starts with 'basic ' (case insensitive), no registry
     // authentication is needed.
@@ -175,7 +177,7 @@ public class RegistryAuthenticator {
             ? serviceMatcher.group(1)
             : registryEndpointRequestProperties.getServerUrl();
 
-    return new RegistryAuthenticator(realm, service, registryEndpointRequestProperties);
+    return new RegistryAuthenticator(realm, service, registryEndpointRequestProperties, userAgent);
   }
 
   private static RegistryAuthenticationFailedException newRegistryAuthenticationFailedException(
@@ -217,14 +219,17 @@ public class RegistryAuthenticator {
   private final String realm;
   private final String service;
   @Nullable private Credential credential;
+  private final String userAgent;
 
   RegistryAuthenticator(
       String realm,
       String service,
-      RegistryEndpointRequestProperties registryEndpointRequestProperties) {
+      RegistryEndpointRequestProperties registryEndpointRequestProperties,
+      String userAgent) {
     this.realm = realm;
     this.service = service;
     this.registryEndpointRequestProperties = registryEndpointRequestProperties;
+    this.userAgent = userAgent;
   }
 
   /**
@@ -295,7 +300,9 @@ public class RegistryAuthenticator {
     try (Connection connection =
         Connection.getConnectionFactory().apply(getAuthenticationUrl(scope))) {
       Request.Builder requestBuilder =
-          Request.builder().setHttpTimeout(JibSystemProperties.getHttpTimeout());
+          Request.builder()
+              .setHttpTimeout(JibSystemProperties.getHttpTimeout())
+              .setUserAgent(userAgent);
 
       if (isOAuth2Auth()) {
         String parameters = getAuthRequestParameters(scope);

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java
@@ -181,7 +181,7 @@ public class RegistryClient {
     // Gets the WWW-Authenticate header (eg. 'WWW-Authenticate: Bearer
     // realm="https://gcr.io/v2/token",service="gcr.io"')
     return callRegistryEndpoint(
-        new AuthenticationMethodRetriever(registryEndpointRequestProperties));
+        new AuthenticationMethodRetriever(registryEndpointRequestProperties, getUserAgent()));
   }
 
   /**

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java
@@ -37,14 +37,15 @@ import javax.net.ssl.KeyManagerFactory;
 import javax.net.ssl.SSLContext;
 
 /** Simple local web server for testing. */
-class TestWebServer implements Closeable {
+public class TestWebServer implements Closeable {
 
   private final boolean https;
   private final ServerSocket serverSocket;
   private final ExecutorService executorService = Executors.newSingleThreadExecutor();
   private final Semaphore threadStarted = new Semaphore(0);
+  private final StringBuffer inputRead = new StringBuffer();
 
-  TestWebServer(boolean https)
+  public TestWebServer(boolean https)
       throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException {
     this.https = https;
     serverSocket = createServerSocket(https);
@@ -52,7 +53,7 @@ class TestWebServer implements Closeable {
     threadStarted.acquire();
   }
 
-  String getEndpoint() {
+  public String getEndpoint() {
     String host = serverSocket.getInetAddress().getHostAddress();
     return (https ? "https" : "http") + "://" + host + ":" + serverSocket.getLocalPort();
   }
@@ -93,12 +94,18 @@ class TestWebServer implements Closeable {
       socket.getOutputStream().flush();
 
       InputStream in = socket.getInputStream();
-      for (int ch = in.read(); ch != -1; ch = in.read()) ;
+      for (int ch = in.read(); ch != -1; ch = in.read()) {
+        inputRead.append((char) ch);
+      }
     }
     return null;
   }
 
   private void ignoreReturn(Future<Void> future) {
     // do nothing; to make Error Prone happy
+  }
+
+  public String getInputRead() {
+    return inputRead.toString();
   }
 }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetrieverTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetrieverTest.java
@@ -41,7 +41,7 @@ public class AuthenticationMethodRetrieverTest {
   private final RegistryEndpointRequestProperties fakeRegistryEndpointRequestProperties =
       new RegistryEndpointRequestProperties("someServerUrl", "someImageName");
   private final AuthenticationMethodRetriever testAuthenticationMethodRetriever =
-      new AuthenticationMethodRetriever(fakeRegistryEndpointRequestProperties);
+      new AuthenticationMethodRetriever(fakeRegistryEndpointRequestProperties, "user-agent");
 
   @Test
   public void testGetContent() {

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -17,8 +17,13 @@
 package com.google.cloud.tools.jib.registry;
 
 import com.google.cloud.tools.jib.configuration.credentials.Credential;
+import com.google.cloud.tools.jib.http.TestWebServer;
+import java.io.IOException;
 import java.net.MalformedURLException;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.security.GeneralSecurityException;
+import org.hamcrest.CoreMatchers;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -170,5 +175,24 @@ public class RegistryAuthenticatorTest {
     Assert.assertEquals(
         new URL("https://somerealm?service=someserver&scope=repository:someimage:scope"),
         registryAuthenticator.getAuthenticationUrl("scope"));
+  }
+
+  @Test
+  public void testUserAgent()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException {
+    try (TestWebServer server = new TestWebServer(false)) {
+      try {
+        RegistryAuthenticator authenticator =
+            RegistryAuthenticator.fromAuthenticationMethod(
+                "Bearer realm=\"" + server.getEndpoint() + "\"",
+                registryEndpointRequestProperties,
+                "Competent-Agent");
+        authenticator.authenticatePush();
+      } catch (RegistryAuthenticationFailedException ex) {
+        // Doesn't matter if auth fails. We only examine what we sent.
+      }
+      Assert.assertThat(
+          server.getInputRead(), CoreMatchers.containsString("User-Agent: Competent-Agent"));
+    }
   }
 }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -35,7 +35,8 @@ public class RegistryAuthenticatorTest {
     registryAuthenticator =
         RegistryAuthenticator.fromAuthenticationMethod(
             "Bearer realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties);
+            registryEndpointRequestProperties,
+            "user-agent");
   }
 
   @Test
@@ -44,7 +45,8 @@ public class RegistryAuthenticatorTest {
     RegistryAuthenticator registryAuthenticator =
         RegistryAuthenticator.fromAuthenticationMethod(
             "Bearer realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties);
+            registryEndpointRequestProperties,
+            "user-agent");
     Assert.assertEquals(
         new URL("https://somerealm?service=someservice&scope=repository:someimage:scope"),
         registryAuthenticator.getAuthenticationUrl("scope"));
@@ -52,7 +54,8 @@ public class RegistryAuthenticatorTest {
     registryAuthenticator =
         RegistryAuthenticator.fromAuthenticationMethod(
             "bEaReR realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties);
+            registryEndpointRequestProperties,
+            "user-agent");
     Assert.assertEquals(
         new URL("https://somerealm?service=someservice&scope=repository:someimage:scope"),
         registryAuthenticator.getAuthenticationUrl("scope"));
@@ -111,17 +114,20 @@ public class RegistryAuthenticatorTest {
     Assert.assertNull(
         RegistryAuthenticator.fromAuthenticationMethod(
             "Basic realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties));
+            registryEndpointRequestProperties,
+            "user-agent"));
 
     Assert.assertNull(
         RegistryAuthenticator.fromAuthenticationMethod(
             "BASIC realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties));
+            registryEndpointRequestProperties,
+            "user-agent"));
 
     Assert.assertNull(
         RegistryAuthenticator.fromAuthenticationMethod(
             "bASIC realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-            registryEndpointRequestProperties));
+            registryEndpointRequestProperties,
+            "user-agent"));
   }
 
   @Test
@@ -129,7 +135,8 @@ public class RegistryAuthenticatorTest {
     try {
       RegistryAuthenticator.fromAuthenticationMethod(
           "realm=\"https://somerealm\",service=\"someservice\",scope=\"somescope\"",
-          registryEndpointRequestProperties);
+          registryEndpointRequestProperties,
+          "user-agent");
       Assert.fail("Authentication method without 'Bearer ' or 'Basic ' should fail");
 
     } catch (RegistryAuthenticationFailedException ex) {
@@ -143,7 +150,7 @@ public class RegistryAuthenticatorTest {
   public void testFromAuthenticationMethod_noRealm() {
     try {
       RegistryAuthenticator.fromAuthenticationMethod(
-          "Bearer scope=\"somescope\"", registryEndpointRequestProperties);
+          "Bearer scope=\"somescope\"", registryEndpointRequestProperties, "user-agent");
       Assert.fail("Authentication method without 'realm' should fail");
 
     } catch (RegistryAuthenticationFailedException ex) {
@@ -158,7 +165,7 @@ public class RegistryAuthenticatorTest {
       throws MalformedURLException, RegistryAuthenticationFailedException {
     RegistryAuthenticator registryAuthenticator =
         RegistryAuthenticator.fromAuthenticationMethod(
-            "Bearer realm=\"https://somerealm\"", registryEndpointRequestProperties);
+            "Bearer realm=\"https://somerealm\"", registryEndpointRequestProperties, "user-agent");
 
     Assert.assertEquals(
         new URL("https://somerealm?service=someserver&scope=repository:someimage:scope"),

